eclass/java-pkg-simple.eclass

```diff
@@ -455,6 +506,30 @@ java-pkg-simple_src_compile() {
 	fi
 }
 
+# @FUNCTION: java-pkg-simple_src_compile
+# @DESCRIPTION:
+# src_compile for multi-jar packages. Wraps java-pkg-simple-mono_src_compile
+# and optionally calla ejavadoc
+java-pkg-simple_src_compile() {
+	java-pkg-simple_mono_src_compile
+
+	# javadoc
+	if has doc ${JAVA_PKG_IUSE} && use doc; then
+		mkdir -p ${apidoc}
+		if [[ -z ${moduleinfo} ]] || [[ ${target#1.} -lt 9 ]]; then
+			ejavadoc -d ${apidoc} \
+				-encoding ${JAVA_ENCODING} -docencoding UTF-8 -charset UTF-8 \
+				${classpath:+-classpath ${classpath}} ${JAVADOC_ARGS:- -quiet} \
+				@${sources} || die "javadoc failed"
+		else
+			ejavadoc -d ${apidoc} \
+				-encoding ${JAVA_ENCODING} -docencoding UTF-8 -charset UTF-8 \
+				${classpath:+--module-path ${classpath}} ${JAVADOC_ARGS:- -quiet} \
+				@${sources} || die "javadoc failed"
+		fi
+	fi
```

Comment:
As we are at it. Should this go in a separate function to? Something like `java-pkg-simple_src_compile_doc`?

Reply:
Or `java-pkg-simple_javadoc`?

Reply:
> As we are at it. Should this go in a separate function, too? Something like `java-pkg-simple_src_compile_doc`?

Putting it in a separate function would require a mass change of all packages (using `java-pkg-simple`) where `ejavadoc` is called within `src_compile()`. I'd prefer doing that later, maybe save it for EAPI 9.